crates/node-bindings/src/file_system/file_system_napi.rs

```diff
@@ -0,0 +1,93 @@
+use std::path::Path;
+use std::path::PathBuf;
+
+use napi::bindgen_prelude::FromNapiValue;
+use napi::threadsafe_function::ThreadsafeFunctionCallMode;
+use napi::Env;
+use napi::JsFunction;
+use napi::JsObject;
+use parcel_filesystem::FileSystem;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+// TODO error handling
+
+pub struct FileSystemNapi {
+  read_file_fn: Box<dyn Fn((PathBuf, String)) -> String + Send + Sync>,
```

Comment:
FYI, using a tuple here will result in the JS function being called with a single array argument rather than separate arguments. We could possibly use a trait to work around this. Unfortunately specialization is not stable in Rust so it's a bit annoying.

```rust
pub trait JsArgs: Serialize {
  fn to_js_args(&self, env: &Env) -> napi::Result<Vec<JsUnknown>> {
    Ok(vec![env.to_js_value(&self)?])
  }
}

impl<A: Serialize, B: Serialize> JsArgs for (A, B) {
  fn to_js_args(&self, env: &Env) -> napi::Result<Vec<JsUnknown>> {
    Ok(vec![env.to_js_value(&self.0)?, env.to_js_value(&self.1)?])
  }
}

impl JsArgs for String {}
impl JsArgs for PathBuf {}
impl JsArgs for bool {}
```